engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/task/ScriptTaskProcessor.java

```diff
@@ -0,0 +1,225 @@
+          .applyOutputMappings(context, element)
+          .flatMap(ok -> stateTransitionBehavior.transitionToCompleted(element, context))
+          .ifRightOrLeft(
+              completed -> stateTransitionBehavior.takeOutgoingSequenceFlows(element, completed),
+              failure -> incidentBehavior.createIncident(failure, context));
+    }
+
+    @Override
+    public void onTerminate(final ExecutableScriptTask element, final BpmnElementContext context) {
+      final var flowScopeInstance = stateBehavior.getFlowScopeInstance(context);
+
+      incidentBehavior.resolveIncidents(context);
+
+      eventSubscriptionBehavior
+          .findEventTrigger(context)
+          .filter(eventTrigger -> flowScopeInstance.isActive())
+          .ifPresentOrElse(
+              eventTrigger -> {
+                final var terminated = stateTransitionBehavior.transitionToTerminated(context);
+                eventSubscriptionBehavior.activateTriggeredEvent(
+                    context.getElementInstanceKey(),
+                    terminated.getFlowScopeKey(),
+                    eventTrigger,
+                    terminated);
+              },
+              () -> {
+                final var terminated = stateTransitionBehavior.transitionToTerminated(context);
+                stateTransitionBehavior.onElementTerminated(element, terminated);
+              });
+    }
+
+    private void triggerProcessEventWithResultVariable(
+        final BpmnElementContext context,
+        final String resultVariableName,
+        final DirectBuffer result) {
+      final DirectBuffer resultVariable = serializeToNamedVariable(resultVariableName, result);
+      eventTriggerBehavior.triggeringProcessEvent(
+          context.getProcessDefinitionKey(),
+          context.getProcessInstanceKey(),
+          context.getElementInstanceKey(),
+          context.getElementId(),
+          resultVariable);
+    }
+
+    private static DirectBuffer serializeToNamedVariable(
+        final String name, final DirectBuffer value) {
+      final var resultBuffer = new ExpandableArrayBuffer();
+      final var writer = new MsgPackWriter();
+      writer.wrap(resultBuffer, 0);
+      writer.writeMapHeader(1);
+      writer.writeString(BufferUtil.wrapString(name));
+      writer.writeRaw(value);
+      return resultBuffer;
+    }
+
+    private Either<Failure, DirectBuffer> evaluateScript(
+        final ExecutableScriptTask element, final BpmnElementContext context) {
+      final var resultOrFailure =
+          expressionProcessor
+              .evaluateAnyExpression(element.getExpression(), context.getElementInstanceKey())
+              .mapLeft(
+                  failure ->
+                      new Failure(
+                          "Expected to evaluate script '%s', but %s"
+                              .formatted(
+                                  BufferUtil.bufferAsString(element.getId()),
+                                  failure.getMessage())));
```

Comment:
:x: We need to set an error type of the failure. For similar cases, we use the `ErrorType.EXTRACT_VALUE_ERROR`.

I tested the behavior of the script task. If the script can't be evaluated successfully (e.g. a referenced variable doesn't exist) then the engine fails to create the incident. I got the following exception:

```
java.lang.NullPointerException: Cannot invoke "java.lang.Enum.ordinal()" because "this.value" is null
	at io.camunda.zeebe.msgpack.value.EnumValue.getEncodedLength(EnumValue.java:79)
	at io.camunda.zeebe.msgpack.property.BaseProperty.getEncodedLength(BaseProperty.java:71)
	at io.camunda.zeebe.msgpack.value.ObjectValue.getEncodedLength(ObjectValue.java:193)
	at io.camunda.zeebe.msgpack.value.ObjectValue.getEncodedLength(ObjectValue.java:131)
	at io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue.getEncodedLength(UnifiedRecordValue.java:26)
	at io.camunda.zeebe.msgpack.UnpackedObject.getLength(UnpackedObject.java:47)
	at io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue.getLength(UnifiedRecordValue.java:20)
	at io.camunda.zeebe.engine.api.records.RecordBatchEntry.createEntry(RecordBatchEntry.java:59)
	at io.camunda.zeebe.engine.api.records.RecordBatch.appendRecord(RecordBatch.java:48)
	at io.camunda.zeebe.streamprocessor.BufferedProcessingResultBuilder.appendRecordReturnEither(BufferedProcessingResultBuilder.java:64)
	at io.camunda.zeebe.engine.api.ProcessingResultBuilder.appendRecord(ProcessingResultBuilder.java:37)
	at io.camunda.zeebe.engine.processing.streamprocessor.writers.ResultBuilderBackedEventApplyingStateWriter.appendFollowUpEvent(ResultBuilderBackedEventApplyingStateWriter.java:40)
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnIncidentBehavior.createIncident(BpmnIncidentBehavior.java:67)
	at io.camunda.zeebe.engine.processing.bpmn.task.ScriptTaskProcessor$ZeebeScriptBehavior.lambda$onActivate$2(ScriptTaskProcessor.java:106)
	at io.camunda.zeebe.util.Either$Left.ifRightOrLeft(Either.java:476)
	at io.camunda.zeebe.engine.processing.bpmn.task.ScriptTaskProcessor$ZeebeScriptBehavior.onActivate(ScriptTaskProcessor.java:101)
	at io.camunda.zeebe.engine.processing.bpmn.task.ScriptTaskProcessor.onActivate(ScriptTaskProcessor.java:48)
	at io.camunda.zeebe.engine.processing.bpmn.task.ScriptTaskProcessor.onActivate(ScriptTaskProcessor.java:29)
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.lambda$processEvent$2(BpmnStreamProcessor.java:113)
	at io.camunda.zeebe.util.Either$Right.ifRightOrLeft(Either.java:381)
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:112)
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.lambda$processRecord$0(BpmnStreamProcessor.java:95)
	at io.camunda.zeebe.util.Either$Right.ifRightOrLeft(Either.java:381)
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:92)
	at io.camunda.zeebe.engine.Engine.process(Engine.java:127)
	at io.camunda.zeebe.streamprocessor.ProcessingStateMachine.lambda$processCommand$3(ProcessingStateMachine.java:261)
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:84)
	at io.camunda.zeebe.streamprocessor.ProcessingStateMachine.processCommand(ProcessingStateMachine.java:257)
	at io.camunda.zeebe.streamprocessor.ProcessingStateMachine.tryToReadNextRecord(ProcessingStateMachine.java:206)
	at io.camunda.zeebe.streamprocessor.ProcessingStateMachine.readNextRecord(ProcessingStateMachine.java:182)
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:92)
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:45)
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:119)
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:106)
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:87)
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:198)
```

Please set the missing error type and create a new test case similar to [BusinessRuleTaskIncidentTest](https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/processing/incident/BusinessRuleTaskIncidentTest.java#L103-L133) (i.e. create incident if evaluation fails + resolve incident).

Reply:
Hi @saig0. Please check this out. 🙇